### PR TITLE
add submission form engine

### PR DESCRIPTION
**Pull request by birdbrained**
Adds submissions form, scheduler to handle display of the form widgets.
Checks conditions in scheduler

### CODE REVIEW

File: client/app/routes/collection/add.js [hunk 1]
```diff
@@ -1,4 +1,124 @@
...
+            sections: [
+                'upload',
+                'disciplines',
+                'basic info',
+                'authors',
+                'submit'
+            ],
+            initial_state: {
+                upload_section: {
+                    state: ['unsaved', 'editing'],
```
Comment (caneruguz) [inline client/app/routes/collection/add.js]: Why is this an array?

File: client/app/controllers/collection/add.js [hunk 2]
```diff
@@ -1,19 +1,127 @@
...
+        // check if the action can fire.
+        const may_fire = check_all(action.conditions);
+        if (!may_fire) { return; }
+
+        // action may fire
+        var return_value = this.get(action.type)(action.parameters);
+        console.log(return_value);
+        if (action.type === 'create_widget') {
+            let wdgs =  this.get('widgets');
+            wdgs.pushObject(return_value);
```
Comment (caneruguz) [inline client/app/controllers/collection/add.js]: It looks like this is where widgets are created. Can they be removed?

File: client/app/routes/collection/add.js (hunk 1 above)
Comment (birdbrained) [inline client/app/routes/collection/add.js]: Anything could have multiple states. Most would only have one, but some have multiple things to keep track of about that parameter. The upload section needs to keep track of whether or not it's saved, and whether or not its being edited.

File: client/app/controllers/collection/add.js (hunk 2 above)
Comment (birdbrained) [inline client/app/controllers/collection/add.js]: I haven't written a remove_widget action yet. The object gets stored in the parameter just below, and it's a reference to the same object. Should be straightforward to delete by parameter.

**Pull request closed without merging**